### PR TITLE
Implement ifMissing value in getEnv()

### PR DESCRIPTION
**Pull request by subsetpark**

### CODE REVIEW

**Comment by Araq** [comment]
Thinking about it, ``os.getEnv`` is available at compiletime so you also need to patch the compiler.

**Comment by Araq** [comment]
Still not enough, grep for 'getEnv' in the compiler's sources.

**Comment by krux02** [comment]
Well I think the changes are an improvement over what currently exists.

File: lib/pure/os.nim [hunk 1]
```diff
@@ -791,18 +791,18 @@ proc findEnvVar(key: string): int =
     if startsWith(environment[i], temp): return i
   return -1
 
-proc getEnv*(key: string): TaintedString {.tags: [ReadEnvEffect].} =
+proc getEnv*(key: string, ifMissing = ""): TaintedString {.tags: [ReadEnvEffect].} =
```
Comment (dom96) [inline lib/pure/os.nim]: Bikeshedding but this argument should be called `default` not `ifMissing`.

**Comment by subsetpark** [comment]
Hey @Araq - which getEnv should I be patching in the compiler? The only other place I see something that looks like that defined, is in `vmops.nim` - and I'm not really sure what that would require.

**Comment by Araq** [comment]
> The only other place I see something that looks like that defined, is in vmops.nim - and I'm not really sure what that would require.

Yes, that one. you need to pass the argument to getEnv.

**Comment by Araq** [comment]
Merged manually.

**Pull request closed without merging**